lib/Modules/ADCharmUtils/ADCharmUtils.psm1

```diff
@@ -101,33 +96,33 @@ function Get-MyADCredentials {
 }
 
 function Get-ActiveDirectoryContext {
-    $blobKey = ("djoin-" + $COMPUTERNAME)
+    if($global:JUJU_AD_RELATION_CONTEXT) {
+        return $global:JUJU_AD_RELATION_CONTEXT
+    }
+    $blobKey = ("djoin-" + [System.Net.Dns]::GetHostName())
     $requiredCtx = @{
-        "already-joined-$COMPUTERNAME" = $null
+        "already-joined-${env:COMPUTERNAME}" = $null
```

Comment:
```powershell
${env:COMPUTERNAME}
```

is all caps on Nano server. For charms such as nova-hyperv it's best to use:

```powershell
[System.Net.Dns]::GetHostName()
```

I am aware this is a dependency of AD charm, but I think it's worth noting gere